HeartbeatLogging/Sources/Heartbeat.swift

```diff
@@ -45,20 +47,20 @@ struct Heartbeat: Codable, Equatable {
   /// various time periods. Because a single heartbeat can help calculate moving averages for multiple
   /// time periods, this property serves to capture all the time periods that the heartbeat can represent in
   /// a moving average.
-  var timePeriods: [TimePeriod] = []
+  let timePeriods: [TimePeriod]
 
   /// Intializes a `Heartbeat` with given `info` and, optionally, a `date` and `version`.
   /// - Parameters:
-  ///   - info: An anonymous piece of information to associate the heartbeat with.
+  ///   - agent: An anonymous piece of information to associate the heartbeat with.
   ///   - date: The date when the heartbeat was recorded. Defaults to the current date.
```

Comment:
Yes, dates are based on device. Foundation's `Date` type represents [A specific point in time, independent of any calendar or time zone.](https://developer.apple.com/documentation/foundation/date#:~:text=a%20specific%20point%20in%20time%2C%20independent%20of%20any%20calendar%20or%20time%20zone.). Given that, I believe traveling across time zones won't matter as dates are compared independent of calendar/timezone. 

I think it's worth adding a unit test that tests for this behavior so I'll do that!

There is an edge case worth calling out though since dates are based on device. If device time is manually changed, the date returned from `Date()` will be distorted based on the actual time IRL. The current implementation is susceptible to this behavior as well though. It's another layer of complexity though for sure though.